src/routes/news/containers/WithCustomNewsProvider.js

```diff
@@ -0,0 +1,309 @@
+export const TUNEWS = 'tunews'
+export const LOCAL = 'local'
+
+const newsTabs = [
+  {
+    type: LOCAL,
+    toggleAttr: 'pushNotificationsEnabled'
+  },
+  {
+    type: TUNEWS,
+    active: activeInternational,
+    inactive: inactiveInternational,
+    toggleAttr: 'tunewsEnabled'
+  }
+]
+
+const NewsTypeIcon = styled.Image`
+  align-self: center;
+`
+const HeaderContainer = styled.View`
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+`
+const TouchableWrapper = styled.TouchableOpacity`
+  margin-top: 17px;
+  margin-bottom: 5px;
+  margin-horizontal: 10px;
+`
+const Loader = styled.ActivityIndicator`
+  margin-top: 15px;
+`
+const LocalTabWrapper: StyledComponent<
+  { isSelected: boolean },
+  ThemeType,
+  *
+> = styled.View`
+  padding-horizontal: 10px;
+  border-radius: 10px;
+  height: 34px;
+  text-align: center;
+  min-width: 110px;
+  align-items: center;
+  justify-content: center;
+  background-color: ${props =>
+    props.isSelected ? props.theme.colors.themeColor : '#959595'};
+`
+
+const LocalText: StyledComponent<{}, ThemeType, *> = styled.Text`
+  font-size: 18px;
+  font-family: ${props => props.theme.fonts.decorativeFontBold}
+  color: ${props => props.theme.colors.backgroundColor};
+`
+export type RouteNotInitializedType = {| status: "routeNotInitialized" |}
+
+export type ProviderRefreshPropType = {|
+  cityModel?: CityModel,
+  navigation: NavigationScreenProp<*>,
+  cityCode: string,
+  language: string
+|}
+
+export type LoadingType = {|
+  status: "loading",
+  innerProps?: {
+    newsId: ?string,
+    selectedNewsType: NewsType,
+    ...ProviderRefreshPropType
+  }
+|}
+
+export type LoadingMoreType<S: {}> = {| status: "loadingMore", innerProps: S |}
+
+export type ErrorType<R: {}> = {|
+  status: "error",
+  message: ?string,
+  code: ErrorCodeType,
+  refreshProps: R | null
+|}
+
+export type ProviderPropType = {|
+  dispatch: Dispatch<StoreActionType>,
+  newsId: ?string,
+  selectedNewsType: NewsType,
+  ...ProviderRefreshPropType
+|}
+
+export type LanguageNotAvailableType = {|
+  status: "languageNotAvailable",
+  availableLanguages: Array<LanguageModel>,
+  changeUnavailableLanguage: (
+    dispatch: Dispatch<StoreActionType>,
+    newLanguage: string
+  ) => void,
+  innerProps: {
+    newsId?: string,
+    selectedNewsType: NewsType,
+    ...ProviderRefreshPropType
+  }
+|}
+
+export type SuccessType<S: {}, R: {}> = {|
+  status: "ready",
+  innerProps: S,
+  refreshProps: R
+|}
+
+export type StatusPropsType<S: {}, R: {}> =
+  | RouteNotInitializedType
+  | LoadingType
+  | LoadingMoreType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>>
+  | ErrorType<R>
+  | LanguageNotAvailableType
+  | SuccessType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>
+
+export type PropsType<
+  S: {
+    ...ProviderPropType
+  },
+  R: {}
+> = {|
+  ...StatusPropsType<S, R>,
+  ...ProviderPropType
+|}
+
+const NewsTypeItem = ({ tab, onItemPress, selectedNewsType, t, theme }) => {
```

Comment:
This component should be extracted to a sparate file as this is quite big.

Reply:
I will move header to another component